js/src/builtin/streams/MiscellaneousOperations.cpp

```diff
@@ -30,6 +30,54 @@ using JS::MutableHandle;
 using JS::ToNumber;
 using JS::Value;
 
+// https://streams.spec.whatwg.org/#transfer-array-buffer
+// As some parts of the specifcation want to use the abrupt completion value,
+// this function may leave a pending exception if it returns nullptr.
+[[nodiscard]] JSObject* js::TransferArrayBuffer(JSContext* cx, JS::Handle<JSObject*> buffer) {
+  MOZ_ASSERT(JS::IsArrayBufferObject(buffer));
+
+  // Step 1.
+  MOZ_ASSERT(!JS::IsDetachedArrayBufferObject(buffer));
+
+  // Step 3 (Reordered)
```

Comment:
Could we add a comment which explains why the steps are reordered -- I think they are reordered because the call to `JS::StealArrayBufferContents` sets the ArrayBuffer's length to 0, which is why we need to get the length before stealing the contents

Reply:
I've reordered to match the latest spec.